flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -686,7 +696,7 @@ public MemoryManager getTaskMemoryManager(AllocationID allocationID) throws Slot
 
 	@Override
 	public void notifyTimeout(AllocationID key, UUID ticket) {
-		checkInit();
+		checkStartedOrClosing();
```

Comment:
I am not sure whether this always holds true. I could imagine the situation in which the `Timeout` in `TimerService` fires and just before it calls `notifyTimeout`, we finish closing the `TaskSlotTable`. In this situation we would fail this check.

Reply:
See answer in the comment to `isValidTimeout`